### ISSUE TITLE
Dropdown Fields Don't Work

### ISSUE THREAD

**Issue by enxaqueca** [yikesinc/yikes-inc-easy-mailchimp-extender#367]
Form's dropdown lists are frozen, meaning they won't expand when clicked on. Being required fields, (I believe that's why) the form won't submit (nor will it disappear or redirect to another page after submission when I  set up these options). To see the issue, please locate the dropdown fields at www.alexandrefeldman.com.br/newsletter/ . The plugin is running on the latest version of WordPress and the Genesis theme.

**Comment by EvanHerman** [comment]
Hi enxaqueca,

Thanks for reporting the issue here. It seems that when we strip slashes to prevent issues with accented characters, something is going wrong.

I am working on getting this issue resolved for you, and I will push out an update and let you know here.


Thanks,
Evan

**Comment by EvanHerman** [comment]
Ok, so I've gone through and added the appropriate markup to ensure that accented characters are escaped and parsed properly on both the front/backend of the site.

I did some testing in a number of language (french, spanish, chinese) to ensure that things work properly.

You can download the latest stable release from the following url:
https://cloudup.com/files/i99vHOTEgfi/download

Or you can head into the staging branch of this repo and inside of the readme file below, click on the download link for the latest stable beta release:
https://github.com/yikesinc/yikes-inc-easy-mailchimp-extender/tree/staging

After downloading and re-uploading the files, could you let us know if things are resolved on your end.


Thanks again!


Evan

**Comment by enxaqueca** [comment]
Evan, I'll download and test them right now. However, this might be a MailChimp issue - I pasted an Embedded Form directly from my MailChimp into another page, www.enxaqueca.com.br/blog/newsletter , and am encountering the same issue. I tried checking Disable All Javascript just to give it a try, but with no success. I'll let you know how it works with the newly downloaded files.

**Comment by enxaqueca** [comment]
Congratulations Evan, the dropdowns are now functional. However, the form won't Hide After Sign Up as per Submission Settings. More importantly, MailChimp hasn't added the data into my List. Should I open a new thread?

**Comment by EvanHerman** [comment]
Hi enxaqueca,

I will look into the form not hiding properly. Is your form set to ajax submissions?

In regards to the data not being added to your list, that could either be due that you haven't accepted the double optin, or things are currently delayed a bit. Give it a little bit and check again. If the data still hasn't hit the list, I can try signing up to that list to see if I get the notification email.


Thanks,
EVan

**Comment by enxaqueca** [comment]
Evan, my form is set to ajax submissions.

Besides, I haven't received a confirmation email for the double opt in (I did look in my spam folder).

**Comment by EvanHerman** [comment]
I am looking into the issue now.

**Comment by EvanHerman** [comment]
It looks like the form on that site is using the MailChimp embed code, not the shortcode provided by our Plugin.

I did some quick testing on my end, set the forms to double opt-in and set them to hide on a succesful signup. From what I can see it looks to be working properly. I did receive the confirmation email, and my forms were hidden after successfully signing up.

Would you mind placing our shortcode back on the page so I can take a look. From your description it sounds like some sort of JavaScript error on the page, which would cause the form to not submit data and the form to not hide after a successful signup.

Another thing that you can test is to set things to non-ajax, just to confirm that data is properly being sent to MailChimp.



Thanks,
Evan

**Comment by enxaqueca** [comment]
Evan, have you looked at www.alexandrefeldman.com.br/newsletter/ ? That's the one with the shortcode provided by the Plugin.

**Comment by EvanHerman** [comment]
Oh my mistake, I clicked on the link in the post above, /blog/newsletter.

So after re-looking into the issue, I can see that there are 3 different javascript errors on the page. Which would explain why the forms ajax functionality is not working.

You have two options to get things working. Your first option is to clean up the .js errors on thepage, which are currently being thrown by jquery.cookie.min.js, and WoocOmmerce cart-fragments.min.js - which is most likely a conflict with either your theme or another plugin you have running.

The other, and quicker option, would be to set the form to submit data as a non-ajax form. This will submit it via an HTTP request and your page will refresh, but regardless of js errors on the page this method will work. In our previous release we didn't provide a toggle, so any user with any sort of js error on the page was having issues. This version we've built in a toggle to allow the users to test out ajax vs non-ajax forms.

Give the non-ajax form a try and see if you get the email and the form gets hidden after submission.



Thanks,
Evan
